disnake/abc.py

```diff
@@ -1862,31 +1862,67 @@ async def fetch_message(self, id: int, /) -> Message:
         data = await self._state.http.get_message(channel.id, id)
         return self._state.create_message(channel=channel, data=data)
 
-    async def pins(self) -> List[Message]:
-        """|coro|
+    def pins(
+        self, *, limit: Optional[int] = 50, before: Optional[SnowflakeTime] = None
+    ) -> ChannelPinsIterator:
+        """Returns an :class:`.AsyncIterator` that enables receiving the destination's pinned messages.
 
-        Retrieves all messages that are currently pinned in the channel.
+        You must have the :attr:`.Permissions.read_message_history` and :attr:`.Permissions.view_channel` to use this.
 
         .. note::
 
             Due to a limitation with the Discord API, the :class:`.Message`
             objects returned by this method do not contain complete
             :attr:`.Message.reactions` data.
 
+        .. versionchanged:: 2.11
+            Return an :class:`.AsyncIterator` to support the discord API changes.
+            ``await channel.pins()`` is deprecated in favor of ``async for msg in channel.pins()``.
+
+        Examples
+        --------
+        Usage ::
+
+            counter = 0
+            async for message in channel.pins(limit=100):
+                if message.author == client.user:
+                    counter += 1
+
+        Flattening to a list ::
+
+            pinned_messages = await channel.pins(limit=100).flatten()
+            # pinned_messages is now a list of Message...
+
+        All parameters are optional.
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The number of pinned messages to retrieve.
+            If ``None``, retrieves every pinned message in the channel. Note, however,
+            that this would make it a slow operation.
+        before: Optional[Union[:class:`.abc.Snowflake`, :class:`datetime.datetime`]]
+            Retrieve pinned messages before this date or message.
+            If a datetime is provided, it is recommended to use a UTC aware datetime.
+            If the datetime is naive, it is assumed to be local time.
+            When passing an :class:`.Object` the API will return messages that were pinned
+            before the creation date of that snowflake. This happens because the API endpoint
+            is paginated based on the ``pinned_at`` timestamp and not the creation
+            timestamp of the snowflake.
```

Comment:
```suggestion
            Retrieve messages pinned before this date or message.
            If a datetime is provided, it is recommended to use a UTC aware datetime.
            If the datetime is naive, it is assumed to be local time.
```
This is a rather lengthy description otherwise, something like this should suffice.